Enable merge option for datasets (phase 1)

Sigmonia opened the pull request:
#### Rationale
Spec: https://docs.google.com/document/d/110fl8qRIk1o9lEgFq8RKFWNW5vo_cYCiWkE1doFq214/
Enable the Merge option for bulk updating Datasets

TODO: (Phase 2) Improve Audit Logging

#### Changes
* Enabled UI option for the Dataset bulk insert page and form
* Passed importOption and auditLogBehavior parameters through to the DataIterator context

Sigmonia committed:
Enable merge option for datasets phase1
- Enable UI option
- Pass through option parameter and DI context options

Sigmonia committed:
Fixed erroneous "No current record found" message for audit details page
Changed success message suffix to "imported" due to enabling merge option

labkey-susanh commented on study/src/org/labkey/study/controllers/DatasetController.java:
```diff
@@ -197,7 +175,7 @@ public ModelAndView getView(DatasetAuditHistoryForm form, BindException errors)
             protected void renderInternal(Object model, PrintWriter out)
             {
-                out.write("<p>No current record found</p>");
+                out.write("<p>No additional details recorded</p>");
```
Any tests that need updating as a result of this text change?

Sigmonia replied:
Not that i could find.

labkey-susanh commented on study/src/org/labkey/study/controllers/DatasetController.java:
```diff
@@ -137,32 +137,6 @@ public ModelAndView getView(DatasetAuditHistoryForm form, BindException errors)
             if (newRecord != null)
             {
                 newData = DatasetAuditProvider.decodeFromDataMap(newRecord);
-                String lsid = newData.get("lsid");
```
Does the removal of this check for the lsid mean we'll never see the details?  Or is the case when we have the details to show handled elsewhere?

Sigmonia replied:
This removes the duplicated block seen on the new inserts. The block starting at Line 172 (old): 146 (change) should continue to show the new row or diff as appropriate.

labkey-susanh replied:
Got it.

Sigmonia committed:
Fixed Created(By) getting changed for rows being updated
- Overrode DatasetSchamaTableInfo.persistRows to return a TableInsertDataIteratorBuilder so columns can be ignored

Added style to prevent unsightly box for help tooltip on import option
Per discussions with PM including DetailedAuditLogging for merge as individual rows w/o diff
- Leaving block to add diff commented out for now as it may impact import performance

Sigmonia commented on study/src/org/labkey/study/model/DatasetDefinition.java:
```diff
@@ -1576,15 +1581,43 @@ public TableInfo getSchemaTableInfo()
+        }
+
         @Override
-        public CaseInsensitiveHashSet skipProperties()
```
Removed as it matches the underlying method

Sigmonia commented on study/src/org/labkey/study/model/DatasetDefinition.java:
```diff
@@ -1576,15 +1581,43 @@ public TableInfo getSchemaTableInfo()
             return m;
         }
 
+        public PHI getMaxContainedPhi()
```
Not changed, just a bad diff.

labkey-susanh commented on study/src/org/labkey/study/model/DatasetDefinition.java:
```diff
@@ -1576,15 +1581,43 @@ public TableInfo getSchemaTableInfo()
+//            {
+//                String lsid = String.valueOf(parameters.get("lsid"));
+//                oldRow = DatasetDefinition.this.getDatasetRow(user, lsid);
+//            }
```
This code is in the other branch (https://github.com/LabKey/platform/tree/20.11_fb_dataset_merge_auditing),  so I don't think this commented out block is necessary.

labkey-susanh reviewed (approved):
As we discussed, let's add the check on auditBehavior in the two addAuditLog methods just in case the methods are called not from the DataIterator.  

Otherwise, looks good.

Sigmonia committed:
Added check for auditTypeBehavior.DETAILED to DatasetSchemaTableInfo.addAuditEvent

Sigmonia committed:
Added auditBehavior check to other addAuditEvent method
